src/main/kotlin/net/ccbluex/liquidbounce/features/command/commands/client/fakeplayer/CommandFakePlayer.kt

```diff
@@ -0,0 +1,362 @@
+import net.minecraft.network.packet.c2s.play.PlayerInteractEntityC2SPacket
+import net.minecraft.network.packet.s2c.play.ExplosionS2CPacket
+import net.minecraft.world.explosion.Explosion
+import net.minecraft.world.explosion.ExplosionBehavior
+import java.math.BigDecimal
+import java.math.RoundingMode
+import java.util.*
+
+/**
+ * Fake Player Command
+ *
+ * Allows you to spawn a client side player for testing purposes.
+ */
+object CommandFakePlayer : Listenable {
+
+    /**
+     * Stores all fake players.
+     */
+    private val fakePlayers: MutableList<FakePlayer> = ArrayList()
+
+    private var recording = false
+    private val snapshots: MutableList<PosPoseSnapshot> = ArrayList()
+
+    private val explosionBehavior: ExplosionBehavior = ExplosionBehavior()
+
+    // the entity ids of fake players shouldn't conflict with real entity ids, so they are negative
+    private var fakePlayerId = -1
+
+    fun createCommand(): Command {
+        return CommandBuilder
+            .begin("fakeplayer")
+            .hub()
+            .subcommand(spawnCommand())
+            .subcommand(removeCommand())
+            .subcommand(clearCommand())
+            .subcommand(startRecordingCommand())
+            .subcommand(endRecordingCommand())
+            .build()
+    }
+
+    private fun spawnCommand(): Command {
+        return CommandBuilder
+            .begin("spawn")
+            .parameter(
+                ParameterBuilder
+                    .begin<String>("name")
+                    .verifiedBy(ParameterBuilder.STRING_VALIDATOR)
+                    .optional()
+                    .build()
+            )
+            .handler { _, args ->
+                checkInGame()
+                spawn(args, false)
+            }
+            .build()
+    }
+
+    private fun removeCommand(): Command {
+        return CommandBuilder
+            .begin("remove")
+            .parameter(
+                ParameterBuilder
+                    .begin<String>("name")
+                    .verifiedBy(ParameterBuilder.STRING_VALIDATOR)
+                    .optional()
+                    .build()
+            )
+            .handler { command, args ->
+                checkInGame()
+
+                if (fakePlayers.isEmpty()) {
+                    throw CommandException(translation("liquidbounce.command.fakeplayer.noFakePlayers"))
+                }
+
+                val name = args.getOrNull(0)?.toString() ?: "FakePlayer"
+
+                val playersToRemove = fakePlayers.filter { fakePlayer -> fakePlayer.name.string == name }
+
+                if (playersToRemove.isEmpty()) {
+                    chat(warning((command.result("noFakePlayerNamed", name))))
+                    chat(regular(command.result("currentlySpawned")))
+                    fakePlayers.forEach { fakePlayer ->
+                        chat(regular("- " + fakePlayer.name.string))
+                    }
+
+                    return@handler
+                }
```

Comment:
Are you sure that the `if`-statement does what you want it to do?

Reply:
Looks good to me.. it should notify the user that the faker play they tried to remove not exists. And print the available fake player.

Reply:
ah yes nvm, for some reason I misread the block